test/unit-tests/handlers/targetHandler.test.js

```diff
@@ -78,5 +111,23 @@ describe('TargetHandler', () => {
       expect(targets.matching.length).to.be.equal(0);
       expect(targets.others.length).to.be.equal(3);
     });
+    it('should give the matching tab for regex Title', async () => {
```

Comment:
Added this test case which will check this logic fails. Thanks for pointing it out. 👍 
`if (
        (targetObject instanceof RegExp && isMatchingRegex(target, targetObject)) ||
        isMatchingUrl(target, targetObject)
      )`